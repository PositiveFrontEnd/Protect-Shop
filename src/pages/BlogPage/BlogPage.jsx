import React from "react";
import "./BlogPage.scss"
import blogImgOne from "/Images/blogImg1.jpg"
import blogImgTwo from "/Images/blogImg2.jpg"
import blogImgThree from "/Images/blogImg3.jpg"
import blogImgFour from "/Images/blogImg4.jpg"

const BlogPage = () => {
    return (
        <div className="blog">
            <div className="container">
                <h3 className="blog__title">Blog</h3>
                <ul className="blog__info">
                    <li>
                        <img className="blog__img" src={blogImgOne} alt="blogOne" />
                        <h4 className="info__title">Mini bags forever </h4>
                        <p className="info__description">Small bags have become very popular this season.
                            They are perfect for parties and weekends when you don't have to wear many things.
                            Such bags can be worn both on the shoulder</p>
                        <p className="info__description">and at the fingertips.</p>
                    </li>
                    <li>
                        <img className="blog__img" src={blogImgTwo} alt="blogtwo" />
                        <h4 className="info__title">Sackbags</h4>
                        <p className="info__description">Sackbags with large fasteners and chains have become popular this season.</p>
                        <p className="info__description">These bags are suitable for many styles, from classics to modern minimalism.</p>
                    </li>
                    <li>
                        <img className="blog__img" src={blogImgThree} alt="blogThree" />
                        <h4 className="info__title">Sackbags</h4>
                        <p className="info__description">Sackbags with large fasteners and chains have become popular this season.</p>
                        <p className="info__description">These bags are suitable for many styles, from classics to modern minimalism.</p>
                    </li>
                    <li>
                        <img className="blog__img" src={blogImgFour} alt="blogFour" />
                        <h4 className="info__title">Mini bags forever </h4>
                        <p className="info__description">Small bags have become very popular this season.
                            They are perfect for parties and weekends when you don't have to wear many things.
                            Such bags can be worn both on the shoulder
                            and at the fingertips.</p>
                    </li>
                </ul>
            </div>
        </div>
    )
}


export default BlogPage