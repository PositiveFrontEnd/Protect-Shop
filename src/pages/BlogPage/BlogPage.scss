.blog {
  margin-top: 0;

  .blog__title {
    margin-top: 47px;
    font-size: 24px;
    font-family: "Noto Sans", sans-serif;
    text-align: center;
    text-transform: uppercase;
  }

  .blog__img {
    width: 100%;
  }

  .blog__info {
    display: flex;
    flex-wrap: wrap;
    gap: 12px;
    justify-content: center;
    margin-top: 40px;
    margin-bottom: 60px;
    row-gap: 30px;

    @media (width >= 768px) {
      margin-top: 80px;
      margin-bottom: 60px;
      row-gap: 100px;
    }

    .info__title {
      margin-top: 24px;
      margin-bottom: 8px;
      color: var(--black);
      font-size: 18px;
      text-transform: uppercase;
    }

    .info__description {
      max-width: 607px;
      font-size: 14px;
    }
  }
}
