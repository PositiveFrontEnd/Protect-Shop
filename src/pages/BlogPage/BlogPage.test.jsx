import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import BlogPage from "./BlogPage";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  BlogPage", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {};

        const store = mockStore(initialState);


        const blogPage = render(
            <Provider store={store}>
                <BrowserRouter>
                    <BlogPage />
                </BrowserRouter>
            </Provider>
        );

        expect(blogPage).toMatchSnapshot();
    });
});
