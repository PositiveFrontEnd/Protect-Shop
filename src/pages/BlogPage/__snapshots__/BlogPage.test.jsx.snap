// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  BlogPage > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="blog"
      >
        <div
          class="container"
        >
          <h3
            class="blog__title"
          >
            Blog
          </h3>
          <ul
            class="blog__info"
          >
            <li>
              <img
                alt="blogOne"
                class="blog__img"
                src="/Images/blogImg1.jpg"
              />
              <h4
                class="info__title"
              >
                Mini bags forever 
              </h4>
              <p
                class="info__description"
              >
                Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder
              </p>
              <p
                class="info__description"
              >
                and at the fingertips.
              </p>
            </li>
            <li>
              <img
                alt="blogtwo"
                class="blog__img"
                src="/Images/blogImg2.jpg"
              />
              <h4
                class="info__title"
              >
                Sackbags
              </h4>
              <p
                class="info__description"
              >
                Sackbags with large fasteners and chains have become popular this season.
              </p>
              <p
                class="info__description"
              >
                These bags are suitable for many styles, from classics to modern minimalism.
              </p>
            </li>
            <li>
              <img
                alt="blogThree"
                class="blog__img"
                src="/Images/blogImg3.jpg"
              />
              <h4
                class="info__title"
              >
                Sackbags
              </h4>
              <p
                class="info__description"
              >
                Sackbags with large fasteners and chains have become popular this season.
              </p>
              <p
                class="info__description"
              >
                These bags are suitable for many styles, from classics to modern minimalism.
              </p>
            </li>
            <li>
              <img
                alt="blogFour"
                class="blog__img"
                src="/Images/blogImg4.jpg"
              />
              <h4
                class="info__title"
              >
                Mini bags forever 
              </h4>
              <p
                class="info__description"
              >
                Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
              </p>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="blog"
    >
      <div
        class="container"
      >
        <h3
          class="blog__title"
        >
          Blog
        </h3>
        <ul
          class="blog__info"
        >
          <li>
            <img
              alt="blogOne"
              class="blog__img"
              src="/Images/blogImg1.jpg"
            />
            <h4
              class="info__title"
            >
              Mini bags forever 
            </h4>
            <p
              class="info__description"
            >
              Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder
            </p>
            <p
              class="info__description"
            >
              and at the fingertips.
            </p>
          </li>
          <li>
            <img
              alt="blogtwo"
              class="blog__img"
              src="/Images/blogImg2.jpg"
            />
            <h4
              class="info__title"
            >
              Sackbags
            </h4>
            <p
              class="info__description"
            >
              Sackbags with large fasteners and chains have become popular this season.
            </p>
            <p
              class="info__description"
            >
              These bags are suitable for many styles, from classics to modern minimalism.
            </p>
          </li>
          <li>
            <img
              alt="blogThree"
              class="blog__img"
              src="/Images/blogImg3.jpg"
            />
            <h4
              class="info__title"
            >
              Sackbags
            </h4>
            <p
              class="info__description"
            >
              Sackbags with large fasteners and chains have become popular this season.
            </p>
            <p
              class="info__description"
            >
              These bags are suitable for many styles, from classics to modern minimalism.
            </p>
          </li>
          <li>
            <img
              alt="blogFour"
              class="blog__img"
              src="/Images/blogImg4.jpg"
            />
            <h4
              class="info__title"
            >
              Mini bags forever 
            </h4>
            <p
              class="info__description"
            >
              Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
